pkg/framework/plugins/defaultevictor/defaultevictor.go

```diff
@@ -0,0 +1,173 @@
+// like filtering, sorting, and other ones that might be relevant in the future
+type DefaultEvictor struct {
+	args        *componentconfig.DefaultEvictorArgs
+	constraints []constraint
+}
+
+// IsPodEvictableBasedOnPriority checks if the given pod is evictable based on priority resolved from pod Spec.
+func IsPodEvictableBasedOnPriority(pod *v1.Pod, priority int32) bool {
+	return pod.Spec.Priority == nil || *pod.Spec.Priority < priority
+}
+
+// HaveEvictAnnotation checks if the pod have evict annotation
+func HaveEvictAnnotation(pod *v1.Pod) bool {
+	_, found := pod.ObjectMeta.Annotations[evictPodAnnotationKey]
+	return found
+}
+
+type DefaultEvictorOptions struct {
+	Priority      *int32
+	NodeFit       bool
+	LabelSelector labels.Selector
+}
+
+// New builds plugin from its arguments while passing a handle
+func New(args runtime.Object, nodeIndexer podutil.GetPodsAssignedToNodeFunc) (framework.EvictorPlugin, error) {
+	defaultEvictorArgs, ok := args.(*componentconfig.DefaultEvictorArgs)
+	if !ok {
+		return nil, fmt.Errorf("want args to be of type defaultEvictorFilterArgs, got %T", args)
+	}
+	ev := &DefaultEvictor{
+		args: defaultEvictorArgs,
+	}
+
+	if ev.args.EvictFailedBarePods {
+		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
+			ownerRefList := podutil.OwnerRef(pod)
+			// Enable evictFailedBarePods to evict bare pods in failed phase
+			if len(ownerRefList) == 0 && pod.Status.Phase != v1.PodFailed {
+				return fmt.Errorf("pod does not have any ownerRefs and is not in failed phase")
+			}
+			return nil
+		})
+	} else {
+		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
+			ownerRefList := podutil.OwnerRef(pod)
+			if len(ownerRefList) == 0 {
+				return fmt.Errorf("pod does not have any ownerRefs")
+			}
+			return nil
+		})
+	}
+	if !ev.args.EvictSystemCriticalPods {
+		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
+			if utils.IsCriticalPriorityPod(pod) {
+				return fmt.Errorf("pod has system critical priority")
+			}
+			return nil
+		})
+
+		if ev.args.PriorityThresholds != nil {
+			ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
+				if IsPodEvictableBasedOnPriority(pod, *ev.args.PriorityThresholds) {
+					return nil
+				}
+				return fmt.Errorf("pod has higher priority than specified priority class threshold")
+			})
+		}
+	}
+	if !ev.args.EvictLocalStoragePods {
+		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
+			if utils.IsPodWithLocalStorage(pod) {
+				return fmt.Errorf("pod has local storage and descheduler is not configured with evictLocalStoragePods")
+			}
+			return nil
+		})
+	}
+	if ev.args.IgnorePvcPods {
+		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
+			if utils.IsPodWithPVC(pod) {
+				return fmt.Errorf("pod has a PVC and descheduler is configured to ignore PVC pods")
+			}
+			return nil
+		})
+	}
+	if ev.args.NodeFit {
```

Comment:
Important to keep https://github.com/kubernetes-sigs/descheduler/issues/640#issuecomment-1216506965 in mind. Once the new plugin is merged, discussing the split will be part of introducing the pre-evictor-filter extension point.

Reply:
Taking note :spiral_notepad: ;)